creator-node/src/services/stateMachineManager/nodeHealthManager.js

```diff
@@ -0,0 +1,266 @@
+const axios = require('axios')
+
+const config = require('../../config')
+const { computeContentNodePeerSet } = require('./utils')
+const { logger } = require('../../logging')
+
+const PEER_HEALTH_CHECK_REQUEST_TIMEOUT_MS = config.get(
+  'peerHealthCheckRequestTimeout'
+)
+const MINIMUM_STORAGE_PATH_SIZE = config.get('minimumStoragePathSize')
+const MINIMUM_MEMORY_AVAILABLE = config.get('minimumMemoryAvailable')
+const MAX_FILE_DESCRIPTORS_ALLOCATED_PERCENTAGE =
+  config.get('maxFileDescriptorsAllocatedPercentage') / 100
+const MINIMUM_DAILY_SYNC_COUNT = config.get('minimumDailySyncCount')
+const MINIMUM_ROLLING_SYNC_COUNT = config.get('minimumRollingSyncCount')
+const MINIMUM_SUCCESSFUL_SYNC_COUNT_PERCENTAGE =
+  config.get('minimumSuccessfulSyncCountPercentage') / 100
+
+// Max number of seconds a primary may be unhealthy for since the first time it was seen as unhealthy
+const MAX_NUMBER_SECONDS_PRIMARY_REMAINS_UNHEALTHY = config.get(
+  'maxNumberSecondsPrimaryRemainsUnhealthy'
+)
+
+/**
+ * Tracks and caches health of Content Nodes.
+ * TODO: Add caching for secondaries similar to how primaries have a threshold of time to remain unhealthy for.
+ */
+class NodeHealthManager {
```

Comment:
this class is the same as [`src/snapbackSM/peerSetManager.js`](https://github.com/AudiusProject/audius-protocol/blob/4c559fd7b2a7e3aa5e4772c5b153260841fe9072/creator-node/src/snapbackSM/peerSetManager.js) EXCEPT:
- no code related to `endpointToSPIdMap` (moved to `nodeToSpIdManager.js` because it should be its own service that runs on a cron)
- no `getNodeUsers()` or `buildReplicaSetNodesToUserWalletsMap()` (moved to `src/services/stateMachineManager/monitoring/utils.js` because they're not related to node health and are only used by the monitoring queue)
- no `computeContentNodePeerSet()` (moved to `src/services/stateMachineManager/utils.js`) because it's more of a one-off util function than something explicitly for node health)

Reply:
awesome, love this improved SOC

should this be inside `monitoring/`?

Reply:
also i think its prob fine to keep `computeContentNodePeerSet` in here since so tied to this internal logic and will only ever be used from NdoeHealthManager
like the param `nodeUserInfoList` is generated by NHM

Reply:
all my off the cuff thoughts, down to discuss

Reply:
* synced offline about how it shouldn't be in `monitoring/` because the reconciliation queue will also need it
* moved computeContentNodePeerSet to a private function in nodeHealthManager (prefixed with underscore)